tests/rptest/tests/self_test_test.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import time
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.admin import Admin
+from ducktape.utils.util import wait_until
+
+
+class SelfTestTest(RedpandaTest):
```

Comment:
It'd be great to add some negative tests. Perhaps a test for what happens if controller leadership changes mid-global-test? And a test where some workers crash mid-local-test?

Reply:
done

Reply:
Would still be great if we could add a test that exercises a leadership change, ensuring that the results from on-going tests are still honored by the new leader.

Feel free to pick that up as in a separate PR though